pkg/check/node_perf.go

```diff
@@ -215,3 +202,27 @@ func (c *CheckNodeDiskPerf) CheckNode(ctx *CheckContext, node *v1.Node, vm *mo.V
 func (c *CheckNodeDiskPerf) FinishCheck(ctx *CheckContext) {
 	return
 }
+
+// checkNodePerf Check to see if the specified node should have its performance check.
+func checkNodePerf(node *v1.Node, mcd MetricCheckDef) bool {
+	checkPerf := true
+	_, isMaster := node.Labels["node-role.kubernetes.io/master"]
+	_, isWorker := node.Labels["node-role.kubernetes.io/worker"]
+
+	if isMaster && mcd.checkMasters == false {
+		klog.V(2).Infof("skipping metric check of [%s] as it is not intended for master nodes", mcd.metricName)
+		checkPerf = false
+	} else if isWorker && mcd.checkWorkers == false {
+		if isMaster && mcd.checkMasters {
```

Comment:
This check looks useless, `if isMaster && mcd.checkMasters == false` was already checked at line 212 

What about decomposing these nested conditiones into something sane. Did I miss anything?

```
if isMaster && mcd.checkMasters {
  return true
}

if isWorker && mcd.checkWorkers {
  return true
}

return false